llvm/lib/Target/SPIRV/SPIRVOperands.cpp

```diff
@@ -0,0 +1,178 @@
+//===-- SPIRVOperands.cpp -  Top level definitions for SPIRV ----*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file contains the implementation for helper mnemonic lookup functions,
+// versioning/capabilities/extensions getters for symbolic/named operands used
+// in various SPIR-V instructions.
+//
+//===----------------------------------------------------------------------===//
+
+#include "SPIRVOperands.h"
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/StringRef.h"
+
+#include <string>
+
+namespace {
+using namespace llvm;
+using namespace OperandCategory;
+using namespace Extension;
+using namespace Capability;
+
+struct SymbolicOperand {
+  ::OperandCategory::OperandCategory Category;
+  uint32_t Value;
+  StringRef Mnemonic;
+  uint32_t minVersion;
+  uint32_t maxVersion;
+};
+
+struct ExtensionEntry {
+  ::OperandCategory::OperandCategory Category;
+  uint32_t Value;
+  ::Extension::Extension ReqExtension;
+};
+
+struct CapabilityEntry {
+  ::OperandCategory::OperandCategory Category;
+  uint32_t Value;
+  ::Capability::Capability ReqCapability;
+};
+
+#define GET_SymbolicOperands_DECL
+#define GET_SymbolicOperands_IMPL
+#define GET_ExtensionEntries_DECL
+#define GET_ExtensionEntries_IMPL
+#define GET_CapabilityEntries_DECL
+#define GET_CapabilityEntries_IMPL
+#include "SPIRVGenTables.inc"
+} // end anonymous namespace
+
+std::string
```

Comment:
I think llvm::StringRef would be more appropriate here and in all the similar usecases below